src/kernels/jupyter/jupyterKernelService.node.ts

```diff
@@ -66,7 +66,8 @@ export class JupyterKernelService {
         resource: Resource,
         @logValue<KernelConnectionMetadata>('id') kernel: KernelConnectionMetadata,
         @logValue<IDisplayOptions>('disableUI') ui: IDisplayOptions,
-        @ignoreLogging() cancelToken: CancellationToken
+        @ignoreLogging() cancelToken: CancellationToken,
+        cannotChangeKernels?: boolean
```

Comment:
For 3rd party API usage, when we fail to start a kernel due to missing dependencies we cannot offer the prompt to change the kernel (controller). As the 3rd party API is using a specific controller

Reply:
To make this possible, we need an owner/creator of kernels.
If a kernel is created by 3rd party API, then we cannot offer the usual options such as changing kernels.
Hence added the notion of `creator` into `IKernel` & the like